src/core/qgsvectorlayercache.cpp

```diff
@@ -267,6 +273,7 @@ void QgsVectorLayerCache::layerDeleted()
 void QgsVectorLayerCache::invalidate()
 {
   mCache.clear();
+  mFullCache = false;
```

Comment:
Isn't this still silently disabling full cache with the potential of introducing performance impacts on invalidate?